From 46cbec326539a6cc5e034b638adaa224cef9c0a3 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Mon, 26 Oct 2020 03:54:53 -0700
Subject: [PATCH 1728/1851] dmaengine: xilinx: frmbuf: Replace AP_READY with
 AP_DONE

commit 87d2d58e2137748748d96a630fd93eb0e2d81123 from
https://github.com/Xilinx/linux-xlnx.git

Use AP_DONE signal instead of AP_READY signal for interrupt generation.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Tested-by: Nayan Bhavsar <nayanb@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/dma/xilinx/xilinx_frmbuf.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/dma/xilinx/xilinx_frmbuf.c b/drivers/dma/xilinx/xilinx_frmbuf.c
index 026d593848df..ee9e08d764ee 100644
--- a/drivers/dma/xilinx/xilinx_frmbuf.c
+++ b/drivers/dma/xilinx/xilinx_frmbuf.c
@@ -1169,7 +1169,7 @@ static void xilinx_frmbuf_reset(struct xilinx_frmbuf_chan *chan)
 static void xilinx_frmbuf_chan_reset(struct xilinx_frmbuf_chan *chan)
 {
 	xilinx_frmbuf_reset(chan);
-	frmbuf_write(chan, XILINX_FRMBUF_IE_OFFSET, XILINX_FRMBUF_IE_AP_READY);
+	frmbuf_write(chan, XILINX_FRMBUF_IE_OFFSET, XILINX_FRMBUF_IE_AP_DONE);
 	frmbuf_write(chan, XILINX_FRMBUF_GIE_OFFSET, XILINX_FRMBUF_GIE_EN);
 }
 
@@ -1206,7 +1206,7 @@ static irqreturn_t xilinx_frmbuf_irq_handler(int irq, void *data)
 		}
 	}
 
-	if (status & XILINX_FRMBUF_ISR_AP_READY_IRQ) {
+	if (status & XILINX_FRMBUF_ISR_AP_DONE_IRQ) {
 		spin_lock(&chan->lock);
 		chan->idle = true;
 		if (chan->active_desc) {
-- 
2.31.1

